src/fuzzfetch/fetch.py

```diff
@@ -335,6 +335,124 @@ def _revision_paths(cls, rev, branch, target_platform):
+                                 help='Download from esr-next')
+        branch_args.add_argument('--try', action='store_const', const='try', dest='branch',
+                                 help='Download from try')
+
+        build_group = self.parser.add_argument_group('Build Arguments')
+        build_group.add_argument('-d', '--debug', action='store_true',
+                                 help='Get debug builds w/ symbols (default=optimized).')
+        build_group.add_argument('-a', '--asan', action='store_true',
+                                 help='Download AddressSanitizer builds.')
+        build_group.add_argument('-t', '--tsan', action='store_true',
+                                 help='Download ThreadSanitizer builds.')
+        build_group.add_argument('--fuzzing', action='store_true',
+                                 help='Download --enable-fuzzing builds.')
+        build_group.add_argument('--coverage', action='store_true',
+                                 help='Download --coverage builds. This also pulls down the *.gcno files')
+        build_group.add_argument('--valgrind', action='store_true',
+                                 help='Download Valgrind builds.')
+
+        test_group = self.parser.add_argument_group('Test Arguments')
+        test_group.add_argument('--tests', nargs='+', metavar='', choices=Fetcher.TEST_CHOICES,
+                                help=('Download tests associated with this build. Acceptable values are: ' +
+                                      ', '.join(Fetcher.TEST_CHOICES)))
+        test_group.add_argument('--full-symbols', action='store_true',
+                                help='Download the full crashreport-symbols.zip archive.')
+
+        misc_group = self.parser.add_argument_group('Misc. Arguments')
+        misc_group.add_argument('-n', '--name',
+                                help='Specify a name (default=auto)')
+        misc_group.add_argument('-o', '--out', default=os.getcwd(),
+                                help='Specify output directory (default=.)')
+        misc_group.add_argument('--dry-run', action='store_true',
+                                help="Search for build and output metadata only, don't download anything.")
+
+        near_group = self.parser.add_argument_group('Near Arguments',
+                                                    "If the specified build isn't found, iterate over " +
+                                                    "builds in the specified direction")
+        near_args = near_group.add_mutually_exclusive_group()
+        near_args.add_argument('--nearest-newer', action='store_const', const=Fetcher.BUILD_ORDER_ASC, dest='nearest',
+                               help="Search from specified build in ascending order")
+        near_args.add_argument('--nearest-older', action='store_const', const=Fetcher.BUILD_ORDER_DESC, dest='nearest',
+                               help="Search from the specified build in descending order")
+
```

Comment:
Part 1/5 of suggestion for comment line 956.

```suggestion
        self.build_is_ns = None  # set to a bool in parse_args if args.build is a taskcluster namespace

```